holmes/config.py

```diff
@@ -148,11 +149,31 @@ def model_post_init(self, __context: Any) -> None:
         self._version = get_version()
         self._holmes_info = fetch_holmes_info()
         self._model_list = parse_models_file(MODEL_LIST_FILE_LOCATION)
-        if ROBUSTA_AI:
+        if self._should_load_robusta_ai():
+            logging.info("Loading Robusta AI model")
             self._model_list["Robusta"] = {
                 "base_url": ROBUSTA_API_ENDPOINT,
             }
 
+    def _should_load_robusta_ai(self) -> bool:
+        if not self.should_try_robusta_ai:
+            return False
+
+        # ROBUSTA_AI were set in the env vars, so we can use it directly
+        if ROBUSTA_AI is not None:
+            return ROBUSTA_AI
+
+        # MODEL is set in the env vars, e.g. the user is using a custom model
+        # so we don't need to load the robusta AI model and keep the behavior backward compatible
+        if "MODEL" in os.environ:
+            return False
+
+        # if the user has provided a model list, we don't need to load the robusta AI model
+        if self._model_list:
```

Comment:
I think, in fact, we can load both model list and robusta AI
Both before and after this change, if the user provides both the ROBUSTA_AI env var, and the model list, both will be loaded.
I think it works ok, but the comment and the check seems redundant

Reply:
Yeah this will still work. The thing i didn't want to impact already exists customer. e.g. they don't have ROBUSTA_AI env var and i would like to check If i should load it or not. On old customers that doesn't working with ROBUSTA_AI it is either set to False or not exist. In the case it didn't exist i check if other env var exists which means they already configured their own model and if not only than i load ROBUSTA_AI